WordPress/gutenberg
# Disable device preview button in pattern/template part/navitation editor

Fixes #65931

## What?

This PR disables the device preview button in the site editor, i.e. in resizable editors, where the device preview button should not be used.

## Why?

Currently, whether to disable the device preview button is determined by whether the entity is nested, i.e., `getEditorSettings().onNavigateToPreviousEntityRecord`.

This condition works in the post editor, but in the site editor, patterns, template parts, and navigation editors are not nested, so the device preview button is available unintentionally.

Resizable Editor and Device Preview currently cannot work together, so we should always disable the device preview  button in a resizable Editor.

## How?

This PR checks the post type, not whether the entity is nested.

## Testing Instructions

Please check the entities below. The ✅ mark indicates the entities whose device preview button will be disabled by this PR.

- Post Editor: The Device Preview button is **enabled**
  - Edit Synced Pattern: The Device Preview button is **disabled**
  - Edit Template: The Device Preview button is **enabled**
- Site Editor: 
  - Template: The Device Preview button is **enabled**
  - ✅ Template Part:  The Device Preview button is **disabled**
  - ✅ Pattern: The Device Preview button is **disabled**
  - ✅ Navigation: The Device Preview button is **disabled**
